test/e2e/lua/dynamic_configuration.go

```diff
@@ -43,7 +44,7 @@ var _ = framework.IngressNginxDescribe("Dynamic Configuration", func() {
 		err := enableDynamicConfiguration(f.KubeClientSet)
 		Expect(err).NotTo(HaveOccurred())
 
-		err = f.NewEchoDeploymentWithReplicas(1)
+		err = f.NewEchoDeploymentWithReplicas(3)
```

Comment:
this is going to slow down test by unnecessarily spinning up 3 replicas for every test where you need that only for the test introduced in this PR